src/model/submissions.nit

```diff
@@ -50,88 +46,32 @@ class Program
 	# * `pending` when `check` is called.
 	# * `success` compilation and tests are fine.
 	# * `error` compilation or tests have issues.
-	var status: String = "submitted"
+	var status: String = "submitted" is writable
 
 	# The name of the working directory.
 	# It is where the source is saved and artifacts are generated.
-	var workspace: nullable String = null
+	var workspace: nullable String = null is writable
 
 	# Object file size in bytes.
 	#
 	# Use only if status == "success".
-	var size_score: nullable Int = null
+	var size_score: nullable Int = null is writable
 
-	# Total execution time in number of instruction.
+	# Total execution time.
 	#
 	# Use only if status == "success".
-	var time_score: nullable Int = null
+	var time_score: Float = 0.0 is writable
```

Comment:
Time can be measured in seconds or in instructions, therefore Int is a bit too broad for this

Another solution could be to measure in milliseconds and keep Int as a measurment unit, I don't know which one you prefer?